contracts/ChildChainGaugeInjectoooor.sol

```diff
@@ -112,44 +102,55 @@ contract ChildChainGaugeInjector is ConfirmedOwner, Pausable, KeeperCompatibleIn
         s_gaugeList = gaugeAddresses;
     }
 
-
-    function setValidatedRecipientList(address[] calldata gaugeAddresses,
+    /**
+     * @notice Validate that all periods are finished, and that the supplied schedule has the right amount of balance to execute on
+     * @notice If everything checks out, update recipient list, otherwise, throw revert
+     * @notice This is griefable by someone transferring funds into the contract,
+     * @notice you can use setRecipientList to set a list without validation
+     * @param gaugeAddresses : list of gauge addresses
+     * @param amountsPerPeriod : list of amount of token in wei to be injected each week
+   */
+    function setValidatedRecipientList(
+        address[] calldata gaugeAddresses,
         uint256[] calldata amountsPerPeriod,
         uint8[] calldata maxPeriods
     ) external onlyOwner {
         address[] memory gaugeList = s_gaugeList;
         // validate all periods are finished
         for (uint256 idx = 0; idx < gaugeList.length; idx++) {
             Target memory target = s_targets[gaugeList[idx]];
-            if (target.periodNumber <= target.maxPeriods) {
-                revert("periods not finished");
+            if (target.periodNumber < target.maxPeriods) {
+                revert PeriodNotFinished(target.periodNumber, target.maxPeriods);
             }
         }
         setRecipientList(gaugeAddresses, amountsPerPeriod, maxPeriods);
 
-        if (!checkBalancesMatch()) {
-            revert("balances don't match");
+        if (!checkExactBalancesMatch()) {
+            revert("balance doesn't match for supplied schedules");
```

Comment:
Suggesting to use custom error still persists as it will follow the convention of this contract